pkg/vpcmodel/groupingGraphical.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package vpcmodel
+
+import (
+	"github.com/np-guard/models/pkg/ipblock"
+)
+
+// for the graphical (html, drawio, svg) representation. In the graph presentation, each node must have all relevant edges.
+// this is not the case in the textual presentation. E.g., a textual presentation may look like:
+// 142.0.64.0/17 ->vsi2
+// 142.0.0.0/16 -> vsi1
+// 0.0.0.0/0 -> vsi3
+// 142.0.64.0/17 should also be connected to vsi2 and vsi3
+// In order to add missing edges, we go over all the endpoints in grouping that present external nodes, and check for containment
+// if external endpoint e1 is contained in external end point e2 then all the "edges" of e2 are added to e1
+func (g *GroupConnLines) consistencyEdgesExternal() {
+	// 1. Gets a map from external endpoints name to their IPs
+	eeNameToIPBlock := getMapToIps(g.GroupedLines)
+	// 2. Gets a map from external endpoints name to their endpoint
+	eeNameToEE := getMapToEPEs(g.GroupedLines)
+	// 3. Gets a map from external endpoint name to all the endpoint it contains
+	containedMap := getContainedEndpointMap(eeNameToIPBlock, eeNameToEE)
+	// 3. Add edges, based on the above map (3)
+	g.addEdgesOfContainingEPs(containedMap)
+}
+
+// gets []*groupedConnLine and returns a map from the string presentation of each endpoint to its ipBlock
+func getMapToIps(grouped []*groupedConnLine) (eeToIPBlock map[string]*ipblock.IPBlock) {
+	eeToIPBlock = map[string]*ipblock.IPBlock{}
+	for _, line := range grouped {
+		addExternalEndpointToMap(line.Src, eeToIPBlock)
+		addExternalEndpointToMap(line.Dst, eeToIPBlock)
+	}
+	return eeToIPBlock
+}
+
+// gets []*groupedConnLine and returns a map from the string presentation of each endpoint to the endpoint element
+func getMapToEPEs(grouped []*groupedConnLine) (eeNameToEE map[string]EndpointElem) {
+	eeNameToEE = map[string]EndpointElem{}
+	for _, line := range grouped {
+		if line.Src.IsExternal() {
+			eeNameToEE[line.Src.Name()] = line.Src
+		} else if line.Dst.IsExternal() {
+			eeNameToEE[line.Dst.Name()] = line.Dst
+		}
+	}
+	return eeNameToEE
+}
+
+func addExternalEndpointToMap(ee EndpointElem, endpointsIPBlocks map[string]*ipblock.IPBlock) {
+	if !ee.IsExternal() {
+		return
+	}
+	_, ok := endpointsIPBlocks[ee.Name()]
+	if ok { // no need to update twice; relevant if the same endpoint is in src and dst of different lines
+		return
+	}
+	endpointsIPBlocks[ee.Name()] = groupedExternalToIPBlock(ee)
+}
+
+func groupedExternalToIPBlock(ee EndpointElem) *ipblock.IPBlock {
+	// EndpointElem must be of type groupedExternalNodes
+	elements := []*ExternalNetwork(*ee.(*groupedExternalNodes))
+	var res = ipblock.New()
+	for _, e := range elements {
+		res = res.Union(e.ipblock)
+	}
+	return res
+}
```

Comment:
```suggestion
func groupedExternalToIPBlock(ee *groupedExternalNodes) *ipblock.IPBlock {
	var res = ipblock.New()
	for _, e := range *ee {
		res = res.Union(e.ipblock)
	}
	return res
}
```
Better enforcement of parameter type

Reply:
done (required change a bit broader than the suggested)